cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServlet.java

```diff
@@ -275,33 +294,42 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
                                            endPoint, userParams.toString());
           setErrorResponse(response, "", errorResp, SC_BAD_REQUEST, wantJSON(request));
         } else {
+          if (EndPoint.postEndpoint().contains(endPoint)) {
+            _requestMeter.get(endPoint).mark();
+          } else {
+            throw new UserRequestException("Invalid URL for POST");
```

Comment:
Why do we need this check here? Can't we move `_requestMeter.get(endPoint).mark();` within `switch-case` to avoid duplicate code here?

Reply:
Thanks for the clarification. How about (1) keeping the `if` part, and (2) removing just the `else` here? Wouldn't switch case already throws an exception if the endpoint is invalid?